src/components/AddToWishlistButton.js

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { addToWishlistInDB, removeFromWishlistInDB } from "../api-calls";
+import { useNotification } from "../contexts/notificationContext";
+import { useUser } from "../contexts/userContext";
+import { useWishlist } from "../contexts/wishlistContext";
+
+const AddToWishlistButton = ({ product }) => {
+
+    const [wishlist, updateWishlist] = useWishlist();
+    const [user] = useUser();
+    const [productInWishlist, setProductInWishlist] = useState(false);
+    const navigate = useNavigate();
+    const { notificationHandler } = useNotification();
+
+    const removeFromWishlist = () => {
+
+        removeFromWishlistInDB(product._id, user);
+        notificationHandler("Removed from wishlist");
+        updateWishlist({ type: "REMOVE_FROM_WISHLIST", payload: product._id });
```

Comment:
Generally reducer function starts with "dispath" for naming convention, but this is also fine.